crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -616,6 +616,44 @@ def f(a: str, /, b: str, c: int = 1, *args, d: int = 2, **kwargs):
         }
     }
 
+    #[test]
+    fn function_parameter_same_name_default() {
```

Comment:
Does this test fail before this PR? Just looking at the test, it doesn't seem to me that this test would be impacted by the bug we are fixing in this PR.

The test I was thinking of would go in `types/infer.rs`, not here, and would assert on the inferred type for the return (or default) expression. The test `local_inference` in that file is an example of an existing test that finds an AST node and then asserts on its type. But like I said, I wasn't necessarily expecting we'd add such a test in this PR.

Unless I'm missing something and this test does actually check for this bug, I would say let's remove this test. If you want to try your hand at adding the above style of test, that's great but not required!